### DISCUSSION TITLE
radial-gradient support

### DISCUSSION THREAD

**Discussion by surjithctly** [tailwindlabs/tailwindcss#2599] [Ideas]
Can tailwindcss support radial-gradient? 

I'd love to see it.

**Comment by adamwathan** [comment]
Hey @surjithctly! You can add radial gradients yourself if you like by adding them to the `backgroundImage` section of your tailwind.config.js file:

https://tailwindcss.com/docs/background-image#customizing

```js
module.exports = {
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--gradient-color-stops))',
      }
    }
  }
}
```

Here's a demo:

https://play.tailwindcss.com/fBVvvF0Phn

**Reply by surjithctly** [reply]
That's cool.

**Reply by ErickPetru** [reply]
Yes, it's nice to have the possibility to create more background images.
But I really think classes like those should already exists by default on Tailwind:

    backgroundImage: () => ({
      'gradient-radial-to-tr': 'radial-gradient(115% 90% at 0% 100%, var(--tw-gradient-stops))',
      'gradient-radial-to-tl': 'radial-gradient(115% 90% at 100% 100%, var(--tw-gradient-stops))',
      'gradient-radial-to-br': 'radial-gradient(90% 115% at 0% 0%, var(--tw-gradient-stops))',
      'gradient-radial-to-bl': 'radial-gradient(90% 115% at 100% 0%, var(--tw-gradient-stops))',
    })

**Reply by simonswiss** [reply]
Radial gradients are fairly niche, and there's always a fine line between what should live in core, and what shouldn't.

If radial gradients become a mainstream design trend, we could likely add these to core. I think for now the best place for these to live is in a custom plugin, which makes it easy for you to bring from project to project if you need them consistently 👍

**Reply by ElwinMusadi** [reply]
> Hey @surjithctly! You can add radial gradients yourself if you like by adding them to the `backgroundImage` section of your tailwind.config.js file:
> 
> https://tailwindcss.com/docs/background-image#customizing
> 
> ```js
> module.exports = {
>   theme: {
>     extend: {
>       backgroundImage: {
>         'gradient-radial': 'radial-gradient(var(--gradient-color-stops))',
>       }
>     }
>   }
> }
> ```
> 
> Here's a demo:
> 
> https://play.tailwindcss.com/fBVvvF0Phn

it's not working in tailwindcss v2.0.3

**Reply by Ziorki** [reply]
@ElwinMusadi 
I have tried configuring it with Tailwind 2.0.3 with no success. Tailwind ^2 uses different variable names for gradients, but changing those names in the config file does not work. The compiler removes or ignores this declaration.
 
Workaround for this is adding CSS class manually into your index.css file  below your tailwind imports
`.bg-radial-gradient {
  background-image: radial-gradient(var(--tw-gradient-stops));
}`

**Reply by adamwathan** [reply]
Here's a working example for Tailwind 2:

https://play.tailwindcss.com/fjBjjbkchW

The playground uses v2.0.2 but this works in v2.0.3 as well.

**Reply by Ziorki** [reply]
Indeed, it does work as @adamwathan said. 

I was trying the same config as pointed in tailwind playgrounds but it didn't work. As it turns out I was missing the postcss-import plugin to process the tailwind.config.js file. After adding it in postcss.config.js it works perfectly.

**Reply by warp9pnt9** [reply]
> Yes, it's nice to have the possibility to create more background images.
> But I really think classes like those should already exists by default on Tailwind:
> 
> ```
> backgroundImage: () => ({
>   'gradient-radial-to-tr': 'radial-gradient(115% 90% at 0% 100%, var(--tw-gradient-stops))',
>   'gradient-radial-to-tl': 'radial-gradient(115% 90% at 100% 100%, var(--tw-gradient-stops))',
>   'gradient-radial-to-br': 'radial-gradient(90% 115% at 0% 0%, var(--tw-gradient-stops))',
>   'gradient-radial-to-bl': 'radial-gradient(90% 115% at 100% 0%, var(--tw-gradient-stops))',
> })
> ```

I am new to TailwindCSS and new to customizations, or plugins.

How would I get the ability to pass the shape (circle, ellipse, or x-y %) and positional (x-y %) parameters, not just color stops?  And what if more than one 'via' is used (e.g. from red, via orange, via black, to blue).  Are such things even possible with TailwindCSS?

**Reply by solamichealolawale** [reply]
> ```js
> 'gradient-radial': 'radial-gradient(var(--gradient-color-stops))',
> ```

Using the demo link, it doesn't work for v3

**Reply by owenduncansnobel** [reply]
Ya this doesn't appear to work for v3..

**Reply by maxzz** [reply]
It works with version 3 like a charm, but you need to use the correct variable name. TailwindCSS 2+ uses the name "var(**--tw-gradient-stops**)" instead of "var(**--gradient-color-stops**)".

**Reply by vltansky** [reply]
Tailwind 3 works:
https://play.tailwindcss.com/kQvzcR8OF8

```
module.exports = {
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      }
    }
  }
}
```

**Reply by tarunpotnuru18** [reply]
@vltansky
 module.exports = {
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      }
    }
  }
}
hey could you please tell me how to convert this to v4

**Reply by wongjn** [reply]
In v4 you don't, it's built in. See https://tailwindcss.com/docs/background-image#adding-a-radial-gradient

**Comment by Everett656** [comment]
Yes, plugin requires Tailwind CSS 1.2 or later. If your project uses an older version of Tailwind, you should install the latest 2.x version of this plugin.

**Reply by polyzen** [reply]
Seems like a phishing link was edited into this comment?

**Reply by RobinMalfait** [reply]
@polyzen thanks for noticing it. I removed the link from the comment.

**Comment by d4h0** [comment]
In case someone missed it: In Tailwind 2.x the CSS variable name seems to have changed.

So now you'd add radial gradients via:

```
module.exports = {
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      }
    }
  }
}
```

**Reply by sebmellen** [reply]
Thank you, took me a bit to figure this out on my own 😆.

**Reply by LuisValgoi** [reply]
thanks for this. worked for me.

**Reply by KehindeAyan** [reply]
Perfecto! Works great here. Thank you.

**Reply by diegocaminor** [reply]
it worked for tailwind version 3 too! 😄

**Reply by dkkazmierczak** [reply]
Works great! Thanks :)

**Comment by marinac-dev** [comment]
Is there a way to set gradient as background color? If I want to use it in combination with mix blend, it doesn't work
For example:
`bg-gradient-radial` is extended `backgroundImage` as shown above.

```html
<div class="bg-gradient-radial from-yellow-400 to-gray-9000">
    <div class="mix-blend-multiply bg-some-svg"></div>
</div>
```

PS: I also tried `bg-blend-multiply` but to no avail

**Comment by techniq** [comment]
It would also be nice to support conic gradients as well.  See https://github.com/tailwindlabs/tailwindcss/discussions/6087 for implementation and example.

**Comment by vltansky** [comment]
Tailwind 3 compatible:

```
module.exports = {
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      }
    }
  }
}
```

https://play.tailwindcss.com/kQvzcR8OF8

**Reply by cesswhite** [reply]
work for me, thanks

**Comment by madsenmm** [comment]
Anyone else having issues with webpack builds failing, when using this exact extend?
```
backgroundImage: {
  'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
}
```

**Comment by eslachance** [comment]
So now how do you create arbitrary `at` values? How would you do `at 75% 25%` to position it 3/4 of the way top-right? This is way beyond the capabilities of any beginner to implement on their own. A helper would be really helpful. 

I tried with just adding a custom color in the config and you can't because `bg-somecolor` doesn't translate to `background-` it translates to `background-color` , which doesn't support radial-gradient.

**Comment by 7iomka** [comment]
> So now how do you create arbitrary `at` values? How would you do `at 75% 25%` to position it 3/4 of the way top-right? This is way beyond the capabilities of any beginner to implement on their own. A helper would be really helpful.
> 
> I tried with just adding a custom color in the config and you can't because `bg-somecolor` doesn't translate to `background-` it translates to `background-color` , which doesn't support radial-gradient.


The same question.
At the moment I use arbitrary className for this case: `[background-image:radial-gradient(50%_50%_at_50%_50%,_#0C2D23_0%,_rgba(12,45,35,0)_100%)]`

**Comment by jim-king-2000** [comment]
What is the easiest way to express this in `Tailwind CSS`? We have to use `style` so far.
```jsx
<div
  className={`absolute w-[calc(50%-1px)] aspect-square rounded-tl-full origin-bottom-right rotate-45 transform-gpu bg-default opacity-50 flex items-center justify-center`}
  style={{
    WebkitMaskImage: 'radial-gradient(circle farthest-side at bottom right, transparent 40%, #000 40%)',
    mask: 'radial-gradient(circle farthest-side at bottom right, transparent 40%, #000 40%)',
  }}
>
  <ChevronUpIcon className='h-6 w-6 -rotate-45 transform-gpu' />
</div>
```

**Comment by alexdsgmoura** [comment]
Hey folks, give this a try!

backgroundImage: {'gradient-radial': 'radial-gradient(circle, var(--tw-gradient-from), var(--tw-gradient-to))'}

**Reply by jim-king-2000** [reply]
It not only a gradient, but also a `mask`. How to express `mask` in Tailwind CSS?

**Comment by alexdsgmoura** [comment]
Create the custom styles you want in the Tailwind CSS configuration file `tailwind.config.js`

**Reply by jim-king-2000** [reply]
I'm quite new to `Tailwind CSS`. Any example for creating a radial gradient mask in `tailwind.config.js`?

**Reply by jim-king-2000** [reply]
By the way, we are writing a translucent [pan control](https://stackoverflow.com/questions/76527796/how-to-make-a-h5-pan-control-in-pure-css).

**Comment by jim-king-2000** [comment]
And how to express multiple gradients in `Tailwind CSS`?

```jsx
<div
  className='flex flex-1 flex-col bg-[length:20px_20px] bg-repeat'
  style={{
    backgroundImage:
      'linear-gradient(to bottom, transparent 19px, rgba(255, 255, 255, 0.1) 20px), linear-gradient(to right, transparent 19px, rgba(255, 
  255, 255, 0.1) 20px)',
  }}
/>
